src/app/main.cpp

```diff
@@ -943,19 +944,22 @@ int main( int argc, char *argv[] )
   QCoreApplication::setAttribute( Qt::AA_DisableWindowContextHelpButton, true );
 #endif
 
-  // Set up an OpenGL Context to be shared between threads beforehand
-  // for plugins that depend on Qt WebEngine module.
-  // As suggested by Qt documentation at:
-  //   - https://doc.qt.io/qt-5/qtwebengine.html
-  //   - https://code.qt.io/cgit/qt/qtwebengine.git/plain/src/webenginewidgets/api/qtwebenginewidgetsglobal.cpp
-#if 0
-  // this is disabled, because it breaks Qt 3D. See
-  // https://interest.qt-project.narkive.com/GYwuMDac/qwebengineview-qsurfaceformat-errors-in-console
-  // https://bugreports.qt.io/browse/QTBUG-60614
-  // https://bugreports.qt.io/browse/QTBUG-60605
+  // Initialize the default surface format for all
+  // QWindow and QWindow derived components
 #if !defined(QT_NO_OPENGL)
-  QCoreApplication::setAttribute( Qt::AA_ShareOpenGLContexts, true );
+  QSurfaceFormat format;
+  format.setVersion( 4, 3 );
+  format.setProfile( QSurfaceFormat::CoreProfile );
```

Comment:
Are you sure it is OK to set `CoreProfile` here? From my investigation so far, if I understand correctly CoreProfile is only allowed on macOS, because macOS does not support CompatibilityProfile (that's a limitation of macOS's OpenGL implementation). But on other systems, QtWebEngine outputs a serious looking warning saying that CoreProfile is unsupported: https://github.com/qt/qtwebengine/blob/dev/src/core/web_engine_context.cpp#L213